Add Web Socket Undertow codestart

Sgitario opened the pull request:
Added code start for web sockets undertow extension.

Sgitario commented:
I verified this example by running jbang:
```
jbang qcd@quarkusio -l java,kotlin -c ./quarkus/examples undertow-websockets-example -t 
```

Note that I had to remove the Qute templates (`base/README.tpl.qute.md` and `base/src/...`) to make the jbang worked. Otherwise, it failed by saying that the `readme-header` was not found. So, let me know if I should include these files back.

gsmet commented:
Tests are failing:
```
Error:  Failures: 
Error:    QuarkusCodestartRunIT.testGradle:78->generateProjectRunTests:152->generateProjectRunTests:175 
expected: 0
but was : 1
Error:    QuarkusCodestartRunIT.testGradleKotlinDSL:85->generateProjectRunTests:152->generateProjectRunTests:175 
expected: 0
but was : 1
```
Killing the rest of the CI.

Sgitario commented:
> Tests are failing:
> 
> ```
> Error:  Failures: 
> Error:    QuarkusCodestartRunIT.testGradle:78->generateProjectRunTests:152->generateProjectRunTests:175 
> expected: 0
> but was : 1
> Error:    QuarkusCodestartRunIT.testGradleKotlinDSL:85->generateProjectRunTests:152->generateProjectRunTests:175 
> expected: 0
> but was : 1
> ```
> 
> Killing the rest of the CI.

Native compilation seems not to be working for undertow (and also resteasy). I've deleted the native test.

gsmet commented on docs/src/main/asciidoc/websockets.adoc:
```diff
@@ -45,8 +45,7 @@ First, we need a new project. Create a new project with the following command:
     -DprojectArtifactId=websockets-quickstart \
-    -Dextensions="undertow-websockets" \
-    -DnoExamples
+    -Dextensions="undertow-websockets"
```
Are we sure we want to do that? It won't be in the way of the rest of the guide?

gsmet replied:
@ia3andy could you check if this makes sense. My understanding is that it's not really a good idea to have the examples in the way of the guides.

Sgitario replied:
As far I can see, there are more guides without `-DnoExamples` that with this property. One example: https://quarkus.io/guides/rest-json
Anyway, let me know if you want this change reverted if you prefer this way.

ia3andy replied:
@Sgitario could you revert this change, we decided not to generate the solution in the guides. At some point we will sync the quickstart by generating it from the codestart (for those which are iso) and give multiple way to get the solution (cmd, quickstart repo, code.quarkus.io) in the Solution section https://quarkus.io/guides/websockets#solution.

Sgitario replied:
Reverted

ia3andy commented:
Thanks for this codestart @Sgitario. I will have a look on Thursday as I will be in PTO tomorrow and my task list if full for today.

ia3andy commented:
> I verified this example by running jbang:
> 
> ```
> jbang qcd@quarkusio -l java,kotlin -c ./quarkus/examples undertow-websockets-example -t 
> ```
> 
> Note that I had to remove the Qute templates (`base/README.tpl.qute.md` and `base/src/...`) to make the jbang worked. Otherwise, it failed by saying that the `readme-header` was not found. So, let me know if I should include these files back.

@Sgitario I just fixed the script because 1.11.1.Final (the default) is not compatible with master, you should call it this way:
```
jbang -Dquarkus.version=999-SNAPSHOT qcd@quarkusio -l java,kotlin -c ./quarkus/examples undertow-websockets-example -t 
```

Of course you should have built quarkus core 999-SNAPSHOT locally first.

Let me know how it goes :)

Sgitario commented:
> > I verified this example by running jbang:
> > ```
> > jbang qcd@quarkusio -l java,kotlin -c ./quarkus/examples undertow-websockets-example -t 
> > ```
> > 
> > 
> > Note that I had to remove the Qute templates (`base/README.tpl.qute.md` and `base/src/...`) to make the jbang worked. Otherwise, it failed by saying that the `readme-header` was not found. So, let me know if I should include these files back.
> 
> @Sgitario I just fixed the script because 1.11.1.Final (the default) is not compatible with master, you should call it this way:
> 
> ```
> jbang -Dquarkus.version=999-SNAPSHOT qcd@quarkusio -l java,kotlin -c ./quarkus/examples undertow-websockets-example -t 
> ```
> 
> you may need to do a:
> 
> ```
> jbang catalog update
> ```
> 
> Of course you should have built quarkus core 999-SNAPSHOT locally first.
> 
> Let me know how it goes :)

Now, it's working perfectly with the qute files. I've added them as the rest of examples.
Thanks a lot!

Sgitario committed:
Add Web Socket Undertow codestart

Sgitario commented on docs/src/main/asciidoc/websockets.adoc:
```diff
@@ -218,6 +217,9 @@ public class ChatTest {
             MESSAGES.add("CONNECT");
+            // Send a message to indicate that we are ready,
+            // as the message handler may not be registered immediately after this callback.
+            session.getAsyncRemote().sendText("_ready_");
```
I had to add this, otherwise the test was not passing. This is the same as done in the quickstart.

ia3andy commented on devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/undertow-websockets-example/java/src/main/java/org/acme/undertowwebsockets/ChatSocket.java:
```diff
@@ -0,0 +1,62 @@
+
+@ServerEndpoint("/chat/{username}")
+@ApplicationScoped
+public class ChatSocket {
```
```suggestion
public class SupersonicChatSocket {
```

ia3andy commented on devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/undertow-websockets-example/java/src/main/resources/META-INF/resources/supersonic-chat.html:
```diff
@@ -0,0 +1,115 @@
+        <nav class="navbar navbar-default navbar-pf" role="navigation">
+                <div class="navbar-header">                  
+                  <a class="navbar-brand" href="/">
+                   <p><strong>>> Quarkus Chat!</strong></p>
```
```suggestion
                   <p><strong>>> Quarkus Supersonic Chat!</strong></p>
```

ia3andy commented on devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/undertow-websockets-example/java/src/main/resources/META-INF/resources/supersonic-chat.html:
```diff
@@ -0,0 +1,115 @@
+
+<head>
+    <meta charset="UTF-8">
+    <title>Quarkus Chat!</title>
```
```suggestion
    <title>Quarkus Supersonic Chat!</title>
```

ia3andy commented on devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/undertow-websockets-example/java/src/test/java/org/acme/undertowwebsockets/ChatSocketTest.java:
```diff
@@ -0,0 +1,55 @@
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+public class ChatSocketTest {
```
```suggestion
public class SupersonicChatSocketTest {
```

ia3andy commented on devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/undertow-websockets-example/codestart.yml:
```diff
@@ -0,0 +1,13 @@
+type: code
+tags: example
+metadata:
+  title: Web Sockets using Undertow
```
```suggestion
  title: WebSockets example using Undertow
```

ia3andy commented on devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/undertow-websockets-example/codestart.yml:
```diff
@@ -0,0 +1,13 @@
+tags: example
+metadata:
+  title: Web Sockets using Undertow
+  description: This example demonstrate how to use Web Sockets to implement an application that send quarks messages
```
```suggestion
  description: Discover WebSockets using Undertow with this cool supersonic chat example. Open multiple tabs to simulate different users.
```

ia3andy reviewed (changes requested):
This is awesome! Thanks!!! I just added a few suggestions..

ia3andy commented:
@Sgitario Thanks!!! could you squash to 1 commit please :)

Sgitario commented:
> @Sgitario Thanks!!! could you squash to 1 commit please :)

Thanks for your review. I've already verified the changes and squashed all the commits in one.

gsmet commented:
I'm not entirely sure we should merge this given @stuartwdouglas just dropped the dependency of websockets with Undertow.

@stuartwdouglas WDYT?

stuartwdouglas commented:
It should be ok, the main difference is that Servlet is no longer pulled in, but this does not really use Servlet so the behavior should be the same.

ia3andy reviewed (approved):
Thanks @Sgitario

ia3andy commented:
@Sgitario could you resolve the conflict please :)

Sgitario commented:
> @Sgitario could you resolve the conflict please :)

Done

ia3andy commented:
Thanks @Sgitario and sorry for the delay ;)
